### PR TITLE
Team5 satisfaction

### PR DESCRIPTION
**Pull request by UMust99**
# Summary

### CODE REVIEW

File: pkg/agents/team5/agent.go [hunk 1]
```diff
@@ -60,17 +69,31 @@ func (a *CustomAgent5) updateSelfishness() {
...
-	if PercentageHP(a) >= 100 {
-		a.satisfaction = 3
+	tmp := int(a.CurrPlatFood())
+	if a.lastFood*120.0/100.0 <= tmp {
+		a.satisfaction += 2
+	} else if a.lastFood*110.0/100.0 <= tmp {
+		a.satisfaction += 1
+	} else if a.lastFood*70.0/100.0 >= tmp {
+		a.satisfaction -= 2
+	} else if a.lastFood*8.0/100.0 >= tmp {
```
[the remark this answers is not in the document]
Comment (UMust99) [reply pkg/agents/team5/agent.go:79]: yeah you're right, it should be 80

File: pkg/agents/team5/agent.go [hunk 2]
```diff
@@ -132,12 +161,14 @@ func (a *CustomAgent5) Run() {
...
 		}
+
 		a.lastMeal = lastMeal
 		if a.lastMeal > 0 {
 			a.daysSinceLastMeal = 0
 		}
-		a.updateSatisfaction()
 		a.hpAfterEating = a.HP()
-		//a.messagingCounter = 0
+		a.updateSatisfaction()
```
Comment (Rhys-J) [inline pkg/agents/team5/agent.go:170]: Does this function call need to be removed?

Comment (UMust99) [reply pkg/agents/team5/agent.go:170]: Yeah, satisfaction should be updated only once at the beginning of the block where the agent tries to eat so this one should be removed

File: pkg/agents/team5/agent.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (UMust99) [reply pkg/agents/team5/agent.go:79]: but the order of checks is ok, if it is less than 70% of the food from previous day(line 77) than we decrease satisfaction by 2 and don't need to check if it's less than 80%

File: pkg/agents/team5/agent.go (hunk 2 above)
Comment (Rhys-J) [reply pkg/agents/team5/agent.go]: Thank you for removing

**Pull request merged**